testsuite/MDAnalysisTests/coordinates/test_imd.py

```
@@ -0,0 +1,418 @@
+        ref.server.send_frames(1, 5)
+
+        reader.add_auxiliary(
+            "lowf",
+            ref.aux_lowf,
+            dt=ref.aux_lowf_dt,
+            initial_time=0,
+            time_selector=None,
+        )
+        reader.add_auxiliary(
+            "highf",
+            ref.aux_highf,
+            dt=ref.aux_highf_dt,
+            initial_time=0,
+            time_selector=None,
+        )
+        yield reader
+        reader.close()
+
+    @staticmethod
+    @pytest.fixture()
+    def transformed(ref):
+        # This will start the test IMD Server, waiting for a connection
+        # to then send handshake & first frame
+        ref.server.handshake_sequence("localhost")
+        # This will connect to the test IMD Server and read the first frame
+        transformed = ref.reader(
+            f"{ref.trajectory}:{ref.server.port}",
+            n_atoms=ref.n_atoms,
+            buffer_size=1 * 1024 * 1024,
+        )
+        # Send the rest of the frames- small enough to all fit in socket itself
+        ref.server.send_frames(1, 5)
+        transformed.add_transformations(
+            translate([1, 1, 1]), translate([0, 0, 0.33])
+        )
+        return transformed
+
+    def test_n_frames(self, ref, reader):
+        pytest.skip("`n_frames` is unknown for IMDReader")
+
+    def test_first_frame(self, ref, reader):
+        # don't rewind here as in inherited base test
+        assert_timestep_almost_equal(
+            reader.ts, ref.first_frame, decimal=ref.prec
+        )
+
+    def test_get_writer_1(self, ref, reader, tmpdir):
+        pytest.skip("No Writer for IMDReader")
+
+    def test_get_writer_2(self, ref, reader, tmpdir):
+        pytest.skip("No Writer for IMDReader")
+
+    def test_total_time(self, ref, reader):
+        pytest.skip("`total_time` is unknown for IMDReader")
+
+    def test_changing_dimensions(self, ref, reader):
+        pytest.skip("IMDReader cannot be rewound")
```

Instead of skipping tests, check that they raise `RuntimeError`. We document the behavior, we need to test that our code behaves as documented.

---

It's also ok to skip tests here (because we need to override them) and then add specific tests below (as already done for, e.g., timeseries).

---

I believe most of these have already been tested in `test_reader_api.py`, since they belong to a higher-level base class?

---

They must be explicitly overriden and skipped, otherwise they fail as StreamReaders only implement a subset of the standard API.

---

Done! Added a `rewind` test